src/eval.rs

```diff
@@ -100,14 +98,63 @@ pub enum DefinitionVariant {
     },
 }
 
+/// Pass through variables needed to make Nix evaluation work inside Nix build. See `initNix`.
+/// If these variables don't exist, assume we're not in a Nix sandbox.
+fn pass_through_environment_variables_for_nix_eval_in_nix_build(command: &mut process::Command) {
+    for variable in [
+        "NIX_CONF_DIR",
+        "NIX_LOCALSTATE_DIR",
+        "NIX_LOG_DIR",
+        "NIX_STATE_DIR",
+        "NIX_STORE_DIR",
```

Comment:
Note for the future: These probably aren't all needed, because we don't do any building, just evaluation. I suspect only `NIX_STATE_DIR` is necessary